From 3e718c9a5892ab2610dac8ff51d31e107b1b00e2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:45 -0400
Subject: [usb] cyberjack: fix buggy integer overflow test

Message-id: <1410887463-20674-34-git-send-email-dzickus@redhat.com>
Patchwork-id: 95113
O-Subject: [RHEL7 PATCH 033/591] USB: cyberjack: fix buggy integer overflow test
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 5287bf726ff8a7353e883b73576710fd53dc88bb
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Sun Oct 6 03:32:53 2013 +0300

    USB: cyberjack: fix buggy integer overflow test

    "old_rdtodo" and "size" are short type.  They are type promoted to int
    and the condition is never true.

    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/cyberjack.c b/drivers/usb/serial/cyberjack.c
index 7814262..6e1b69d 100644
--- a/drivers/usb/serial/cyberjack.c
+++ b/drivers/usb/serial/cyberjack.c
@@ -279,7 +279,7 @@ static void cyberjack_read_int_callback(struct urb *urb)
 
   old_rdtodo = priv->rdtodo;
 
-  if (old_rdtodo + size < old_rdtodo) {
+  if (old_rdtodo > SHRT_MAX - size) {
    dev_dbg(dev, "To many bulk_in urbs to do.\n");
    spin_unlock(&priv->lock);
    goto resubmit;
-- 
1.7.1